engine/src/mbliphonemail.mm

```diff
@@ -263,7 +263,8 @@ static void compose_mail_prewait(void *p_context)
 			else
 				t_name = MCStringConvertToAutoreleasedNSString(ctxt -> attachments[i] . name);
 				
-			[ctxt -> dialog addAttachmentData: t_data mimeType: t_type fileName: t_name];
+            if (t_data.length != 0)
```

Comment:
I'm not sure its the data being zero length which is the problem - addAttachmentData:mimeType:fileName requires that the data, mimeType and fileName are all non-nil; so I think the bug is that if (ctxt->attachments[i].name == nil) then t_name = @"";